### ISSUE TITLE
bug: infinite loop loading buysellads

### ISSUE THREAD

**Issue by mrsvt** [Semantic-Org/Semantic-UI-React#3447]
### Steps
A clear and concise description of steps to reproduce the problem.
1. Open https://react.semantic-ui.com/elements/button/#types-button
2. Open Chrome devtools
3. Check `elements` or `network` tab to see the infinite loop

### Expected Result
The result that you expected.

### Actual Result
The actual result that happened 💣
![feb-19-2019 23-36-54](https://user-images.githubusercontent.com/28084599/53031382-53b00600-349f-11e9-8edb-0546650c678a.gif)
Added in https://github.com/Semantic-Org/Semantic-UI-React/blob/dda01825f8fec7c0ccd4491145d9a34b3d475207/templates/src-components-ComponentDoc.1dc0af3c.js

### Version
0.85.0

### Testcase
[Fork, update, and replace this pen to show the bug]:
https://codesandbox.io/s/2l3n74j9y

**Comment by layershifter** [comment]
Facing this, too.

/cc @levithomason

**Comment by levithomason** [comment]
I am not able to reproduce this problem using:

- 15" MacBook Pro (2017), macOS High Sierra
- Chrome Version 72.0.3626.121 (Official Build) (64-bit)
- Semantic UI React docs v0.85.0

I've also tried this:

- locally and on production
- with and without ad blockers
- in examples which repeatedly re-render (possibly triggering new ad fetches)

Any more info I can go on @layershifter / @sayuti-daniel?

**Comment by layershifter** [comment]
I can reproduce it locally and on production. It happens on wrong response from `srv.buysellads.com`.

#### Wrong response 

```js
_handleNativeJSON({"ads":[{"active":"1","ad_via_link":"https://www.buysellads.com/?utm_source=react-semantic-ui-com-native\u0026utm_medium=ad_via_link\u0026utm_campaign=in_unit\u0026utm_term=custom","external_id":"1261","height":"0","longimp":"T3JTTXBATTTTTTT3G7XMYTTTTTTT4XIXZETTTTTTT3XFOUTTTTTTT","noincrement":"1","rendering":"custom","statimp":"//srv.buysellads.com/ads/imp/x/GTND42QMC6AD627ECE74YKQMCWYIE2Q7CVYD4Z3JCWBI453MF67DTKQKC6BIPKQYCWAI6K3EHJWNBADLKM7UCBZG2Y","timestamp":"1551859592","width":"0","zoneid":"32354","zonekey":"CK7DC2QW"},{}]});
```

### Valid response

```js
_handleNativeJSON({"ads":[{"active":"1","ad_via_link":"https://www.buysellads.com/?utm_source=react-semantic-ui-com-native\u0026utm_medium=ad_via_link\u0026utm_campaign=in_unit\u0026utm_term=custom","backgroundColor":"#58409B","backgroundHoverColor":"#58409B","bannerid":"86364","callToAction":"Start Now","company":"Facebook Audience Network","companyTagline":"Discover monetization that puts users ﬁrst","creativeid":"183812","ctaBackgroundColor":"#54c7EC","ctaBackgroundHoverColor":"#54c7EC","ctaTextColor":"#FFFFFF","ctaTextColorHover":"#FFFFFF","description":"Get quality ads for the great user experience your app deserves, plus tools and insights for smarter monetization.","evenodd":"0","external_id":"8237","height":"0","i":"0","image":"https://cdn4.buysellads.net/uu/1/41312/1548894290-AN_Icon.jpg","logo":"https://cdn4.buysellads.net/uu/1/41312/1548894295-AN_Logo_White.png","longimp":"T3JTTXBATTTTTT4AHRXHYTTTTTTTQKDRZETTTTTTT3XFOUTTTTTTT","longlink":"THJTTXBATTTTTT4AHRXHYTTTTTTTQKDRZETTTTTTT3XFOUTTTTTTTBDW5JYFC5JGHRENVZI35JLCBR35K2UNBZSDHRWNB77SKHLFKGQJC67DTKJLCTYIC5QECKYIL5SHKKOI4KQLCT7D6KQMCY7IL27NFTRNCZI3FWBD5KQIF2Y4AYQRCWADPK7SK6UC4GQYCV7I55SIH7EIBKQSKH7LAYQRFT7DC27S2HLCKYQRF2WNV7I3ZRNC4R3RF2EN4BZ6FWRN4YZQKJGNYYZWFWRNCYZD23LIB5SJ53JDBKQSFRWDBBDW5JYFC5JGHRSU5R75K7ENC7Z42RUNNZID2RNLOYZMKJWNPSIDKMUCPRDU2R7CNZSE5QUNVRZD5J7LOSSNKM7FKBZH5AXUPRD2ZR7NORZLK2LIBKPVCVYD45PXVHFHCIP4ZR3N4AI42RU4KRZW2MGNB7Z62HBNBGP62H7UTSD3G6IFPRD2ZRJN4SZEKHWN5SJRVQQW4UB5KHBCLRIHZRQ6OM7C5MBCBUSD2RUFV7Z55EONAYZW2HSCP","pixel":"https://ad.atdmt.com/i/img;adv=11202201390319;ec=11202215639768;c.a=4713;p.a=5853;a.a=8237;s.a=1;as.a=8237;idfa=;idfa_lat=;aaid=;aaid_lat=;cache=","rendering":"custom","statimp":"//srv.buysellads.com/ads/imp/x/GTND42QMC6AD627ECE74YKQMCWYI527ICW7I6Z3JCWBI427WCTAICKJKC6BIPKQYCWAI6K3EHJWNBADLKM7UCBZG2Y","statlink":"//srv.buysellads.com/ads/click/x/GTND42QMC6AD627ECE74YKQMCWYI527ICW7I6Z3JCWBI427WCTAICKJKC6BIPKQYCWAI6K3EHJNCLSIZ","textColor":"#FFFFFF","textColorHover":"#FFFFFF","timestamp":"1551860180","title":"Discover monetization that puts users ﬁrst.","transparentLogo":"https://cdn4.buysellads.net/uu/1/41312/1548894300-AN_Logo_White.png","width":"0","zoneid":"32354","zonekey":"CK7DC2QW"},{}]});
```

---

```js
      const sanitizedAd = json.ads
        .filter(ad => Object.keys(ad).length > 0)
        .filter(ad => !!ad.statlink)
        .filter(Boolean)[0]
      debug('handleNativeJSON sanitizedAd', sanitizedAd)
```

`Wrong response` doesn't contain `ad.statlink`, so `sanitizedAd` will be empty and will start `getAd()` again and again. May be small timeout can resolve this issue?

**Comment by mrsvt** [comment]
Mine keep getting wrong response as mentioned by @layershifter. It only happened whenever the editor is opened.

FYI, I'm using:
* 15" MacBook Pro (2016), macOS Mojave
* Chrome Version 72.0.3626.121 (Official Build) (64-bit)
* Semantic UI React docs v0.85.0

**Comment by levithomason** [comment]
Still not able to reproduce this here 😕 Any ideas how you're getting repeatedly bad responses from carbon?  Ad blocker maybe?  Possible request hijacking in an extension?

The logic in place is that if an ad is served without an actionable link (supposed to be extremely rare), then fetch another one.  I just can't get a response from carbon that does not include a `statlink` at all.  Even if I add a debugger at that line, it is never triggered when browsing the doc site since no ads come back without a statlink.

**Comment by levithomason** [comment]
Spoke with the ad service, some regions don't produce proper ads.  I've pushed a fix.

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.86.0`.
